codegen/src/lib.rs

```diff
@@ -20,12 +20,43 @@ use std::io::{Read, Seek};
 use std::path::PathBuf;
 use std::str::{from_utf8, FromStr};
 
-use include_flate_compress::{apply_compression, CompressionMethod};
+use include_flate_compress::{apply_compression, compression_ratio, CompressionMethod};
 use proc_macro::TokenStream;
 use proc_macro2::Span;
 use proc_macro_error::{emit_warning, proc_macro_error};
 use quote::quote;
-use syn::{Error, LitByteStr};
+use syn::{Error, LitByteStr, LitInt, Token};
+
+/// This macro evaluates to `true` if the file should be compressed, `false` otherwise, at compile time.
+/// Useful for conditional compilation without any efforts to the runtime.
+///
+/// Please note that unlike the macro names suggest, this macro does **not** actually compress the file.
```

Comment:
If this is only expected for internal use, should we `#[doc(hidden)]` the re-export? And shall we rename this to `should_deflate`, since this does not actually do the imperative `deflate` as the name suggests?